src/components/app/index.tsx

```diff
@@ -25,33 +66,131 @@ export default class App extends Component<Props, State> {
     }
   }
 
+  getEncoderName(type: ImageType, options: CodecOptions) {
+    const name = EncoderNames[type];
+    if (typeof name === 'function') return name(options);
+    return name;
+  }
+
+  @bind
+  setLeftType(leftType: ImageType) {
+    this.setState({ leftType, leftOptions: {} });
+    this.optionsUpdated();
```

Comment:
Considering we see this pattern a lot, should we make a second decorator where a function will just return the partial update passed to `setState()` and that calls `optionsUpdated()` implicitly?

Not sure if making this DRY is worth it, just a thought. @kosamari @jakearchibald ?

Reply:
I think it's too early for that. We'll be splitting this up so that a change to the left doesn't result in a reencode on the right.

However, @developit, this looks like a case where we want to look at state before/after, and set additional state as a result. As in, "If left's options are different, then update `leftImg`". Is this a common pattern?

We could either split into `leftOptionsUpdated` and `rightOptionsUpdate`, or give `optionsUpdated` the ability to know what's changed.

Reply:
(Either way, I don't think a decorator is saving us much, it's just moving the call up a couple of lines up)

Reply:
I had a few different implementations of this, including the abstracted "setter with optionsUpdated", a componentDidUpdate() hook with a comparison, etc.  TBH I don't care which we go with, these were just the most explicit.